backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -231,7 +233,51 @@ private Entry FromLexEntry(ILexEntry entry)
             LexemeForm = FromLcmMultiString(entry.LexemeFormOA.Form),
             CitationForm = FromLcmMultiString(entry.CitationForm),
             LiteralMeaning = FromLcmMultiString(entry.LiteralMeaning),
-            Senses = entry.AllSenses.Select(FromLexSense).ToList()
+            Senses = entry.AllSenses.Select(FromLexSense).ToList(),
+            ComplexForm = ToComplexForm(entry)
+        };
+    }
+
+    private ComplexForm? ToComplexForm(ILexEntry entry)
+    {
+        var complexEntryRef = entry.ComplexFormEntryRefs.SingleOrDefault();
```

Comment:
I'm impressed that you found this property. I'm guessing it's a good one to use even though I find it somewhat confusing.
I was also confused by the fact that your property is called "ComplexForm", while it contains the contents of the "Components" field in FLEx. But, ultimately I like that decision as well.
It also seems like some of the ComplexForm properties inLibLCM get used for Variants as well. There's some weird overloading there that's not entirely intuitive.

Perhaps you went down all those thought paths as well 😆.

To find complex forms of an entry we'd have to query for entries that use it as a component, right? So, it would be an additional query unless we add more redundancy to the modal. I'm assuming we can do that fairly efficiently?

All that to say that this looks good to me 😉. But, I think we need a professional opinion.

Reply:
yeah so my understanding is that LCM models Complex forms and variants like this:
```js
{ //entry
    LexemForm: {}
    //...
    EntryRefsOS: [
        {
            RefType: "ComplexForm",
            ComponentLexemesRS: []
        },
        {
            RefType: "Variant",
            ComponentLexemesRS: []
        }
    ]
}
```

so the components field when the ref type is a complex form, gets mapped to the components field. However if the ref type is variant then the component field gets mapped to the variant of field.

Maybe I'll add variants while I'm at it since I understand it today.